staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -91,7 +94,7 @@ type Serializers struct {
 // NewRESTClient creates a new RESTClient. This client performs generic REST functions
 // such as Get, Put, Post, and Delete on specified paths.  Codec controls encoding and
 // decoding of responses from the server.
-func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ContentConfig, maxQPS float32, maxBurst int, rateLimiter flowcontrol.RateLimiter, client *http.Client) (*RESTClient, error) {
+func NewRESTClient(baseURL *url.URL, versionedAPIPath string, config ContentConfig, maxQPS float32, maxBurst int, rateLimiter flowcontrol.RateLimiter, client *http.Client, bearerTokenSource BearerTokenSource) (*RESTClient, error) {
```

Comment:
Probably best to not break the function signature. What's the code path from kubeconfig -> RESTClient. Can we just modify the RESTClient later?